File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
 function isAlphaNum (c) {
    return (
        (c >= 'a' && c <= 'z') ||
        (c >= 'A' && c <= 'Z') ||
        (c >= '0' && c <= '9')
    )
 }
var isPalindrome = function(s) {
    let fr = 0;
    let bc = s.length - 1;

    while (fr < bc) {
        while (fr < bc && !isAlphaNum(s[fr])) {
            fr++
        }
        while (fr < bc && !isAlphaNum(s[bc])) {
            bc--
        }

        if (s[fr].toLowerCase() !== s[bc].toLowerCase()) {
            return false
        }
        fr++;
        bc--
    }    

    return true
};

File: 125-valid-palindrome/valid-palindrome.go
func isPalindrome(s string) bool {
    fr := 0
    bc := len(s) - 1;

    for fr < bc {
        for fr < bc && !isAlphaNum(rune(s[fr])) {
            fr++
        }

        for fr < bc && !isAlphaNum(rune(s[bc])) {
            bc--
        }

        if unicode.ToLower(rune(s[fr])) != unicode.ToLower(rune(s[bc])) {
            return false
        }
        
        fr++
        bc--
    }
    return true
}

func isAlphaNum(c rune) bool {
    return (c >= 'a' && c <= 'z') ||
         (c >= 'A' && c <= 'Z') ||
         (c >= '0' && c <= '9')
}